src/main/java/com/conveyal/datatools/manager/models/transform/ReplaceFileFromStringTransformation.java

```diff
@@ -0,0 +1,57 @@
+package com.conveyal.datatools.manager.models.transform;
+
+import com.conveyal.datatools.common.status.MonitorableJob;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * This feed transformation will replace a file in the target zip (table) with the provided csv data.
+ */
+public class ReplaceFileFromStringTransformation extends ZipTransformation {
+
+    public static ReplaceFileFromStringTransformation create(String csvData, String table) {
+        ReplaceFileFromStringTransformation transformation = new ReplaceFileFromStringTransformation();
+        transformation.csvData = csvData;
+        transformation.table = table;
+        return transformation;
+    }
+
+    @Override
+    public void transform(FeedTransformTarget target, MonitorableJob.Status status) {
+        // TODO: Refactor into validation code?
```

Comment:
As I mentioned in `FeedTransformTarget`, you could only accept the zip version of it, with the checks embedded within it.